src/bundle/Controller/RoleController.php

```diff
@@ -188,6 +196,50 @@ public function createAction(Request $request): Response
         ]);
     }
 
+    /**
+     * @param \Symfony\Component\HttpFoundation\Request $request
+     * @param \eZ\Publish\API\Repository\Values\User\Role $role
+     *
+     * @return \Symfony\Component\HttpFoundation\Response
+     */
+    public function copyAction(Request $request, Role $role): Response
+    {
+        $this->denyAccessUnlessGranted(new Attribute('role', 'create'));
+        $form = $this->formFactory->copyRole(
+            new RoleCopyData($role)
+        );
+        $form->handleRequest($request);
+
+        if ($form->isSubmitted()) {
+            $result = $this->submitHandler->handle($form, function (RoleCopyData $data) {
+                $roleCopyStruct = $this->roleCopyMapper->reverseMap($data);
+                $role = $this->roleService->copyRole($data->getClonedRole(), $roleCopyStruct);
+
+                $this->notificationHandler->success(
+                    $this->translator->trans(
```

Comment:
This property does not exist.
(getting "Notice: Undefined property: EzSystems\EzPlatformAdminUiBundle\Controller\RoleController::$translator" after copying a role).